pkg/detector/detector.go

```diff
@@ -523,7 +523,7 @@ func (d *ResourceDetector) ApplyClusterPolicy(object *unstructured.Unstructured,
 					"try again later after binding is garbage collected, see https://github.com/karmada-io/karmada/issues/2090")
 			}
 			// Just update necessary fields, especially avoid modifying Spec.Clusters which is scheduling result, if already exists.
-			bindingCopy.Labels = binding.Labels
+			bindingCopy.Labels = util.DedupeAndMergeLabels(bindingCopy.Labels, binding.Labels)
```

Comment:
Maybe we don't need to modify this line, because we don't currently support cluster scope resources for the dependency distribution feature.

Reply:
In my opinion, this is not just for the dependency distribution feature, but customized labels as well.  
For example, if we have a webhook which would patch some custmized info into the labels of resourcebindings to group them, after I scale the workload once, the info would be lost.  
This might not be a good example, but I think even if it is a label for temporary usage, we shouldn't abandon them.

Reply:
Do you think we should add some E2E to protect this case?

Reply:
Humm......I think it is nice to have.  
I hope to help add some.  
May I commit them in this PR? Or open a new one?

Reply:
Sounds good to me.

Reply:
Thanks~